pkg/k8shandler/logstore.go

```diff
@@ -246,6 +247,10 @@ func (cr *ClusterLoggingRequest) newElasticsearchCR(elasticsearchName string, ex
 
 		esNodes = append(esNodes, dataNode)
 
+		// if the cluster is set to be redeployed, keep the status
+		if len(existing.Status.Nodes) > 0 && existing.Status.Nodes[0].UpgradeStatus.ScheduledForCertRedeploy == v1.ConditionTrue {
```

Comment:
why keep a specific status and not the entirety?

Reply:
is there a way we can ensure that CLO doesn't touch the status field for the es CR at all?
ideally it would only update the spec for an update

Reply:
If so, we probably need only to update the Spec of the existing CR, instead of creating a new one.

Reply:
why do we need to conditionally keep the status instead of just always keeping it?